# The following patch is a workaround for an issue in building
# modules for python 3.6 on i686 [1].
#   The situation occurs at testing to load just-built modules
#     /usr/bin/python3.6 -c 'import xapian'
#     /usr/bin/python3.6 -O -c 'import xapian'
#   And weirdly, only first runs will crash.
#   The issue might be caused by BLODA, and I couldn't figure out
#   an exact reason, yet.
#   So, dummy runs are added just after building and rebasing modules.
# [1]: https://sourceware.org/pipermail/cygwin-apps/2021-April/041196.html

--- origsrc/xapian-bindings-1.4.18/python3/Makefile.am
+++ src/xapian-bindings-1.4.18/python3/Makefile.am
@@ -92,6 +92,9 @@
 ## ksh requires a path on the sourced file.
 	. ./libtoolconfig.tmp; cp $$objdir/_xapian$(PYTHON3_SO) xapian
 	rm -f libtoolconfig.tmp
+	rebase -sO $@
+	-$(PYTHON3) -c 'import xapian'    # Without this, weirdly, it might fail to build for python 3.6 on i686
+	-$(PYTHON3) -O -c 'import xapian' # Without this, weirdly, it might fail to build for python 3.6 on i686
 
 CLEANFILES = \
     xapian/_xapian$(PYTHON3_SO) \
